unit/util/range.cpp

```diff
@@ -42,16 +50,25 @@ SCENARIO("range tests", "[core][util][range]")
     }
     THEN("Filter using lengths")
     {
-      auto filtered_range = make_range(list).filter(
+      const auto filtered_range = make_range(list).filter(
         [&](const std::string &s) { return s.length() == 4; });
       auto it = filtered_range.begin();
       REQUIRE(*it == "cdef");
       ++it;
       REQUIRE(it == filtered_range.end());
     }
+    THEN("A const filter iterator can mutate the input collection.")
```

Comment:
Suggest rephrasing, since "const filter iterator" sounds an awful lot like a `const_iterator`, which this is not